app/models/mhv_metrics_unique_user_event.rb

```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+# Unique User Metrics event tracking model for MHV Portal
+#
+# This model tracks unique user events for analytics purposes. Each record represents
+# the first time a specific user performed a specific event (e.g., viewed MHV landing page,
+# accessed secure messages, etc.).
+#
+# Design notes:
+# - Uses compound unique index (user_id, event_name) to ensure one record per user per event
+# - No foreign key constraint on user_id for performance and historical data preservation
+# - Includes Redis caching to minimize database reads for duplicate event checks
+class MHVMetricsUniqueUserEvent < ApplicationRecord
+  include RedisCaching
+
+  # Cache configuration
+  REDIS_CONFIG_KEY = REDIS_CONFIG[:unique_user_metrics]
+  CACHE_NAMESPACE = 'unique_user_metrics'
+  CACHE_TTL = REDIS_CONFIG_KEY[:each_ttl]
+
+  # Configure Redis caching for duplicate event checks
+  redis_config REDIS_CONFIG_KEY
+
+  # Active Record Validations
+  validates :user_id, presence: true
+  validates :event_name, presence: true, length: { maximum: 50 }
+
+  # Class methods for event logging and checking
+
+  # Check if a specific event has already been logged for a user
+  #
+  # @param user_id [String] UUID of the user
+  # @param event_name [String] Name of the event to check
+  # @return [Boolean] true if event exists, false otherwise
+  def self.event_exists?(user_id:, event_name:)
```

Comment:
Note: This will probably never be used in code, but will be useful for testing this feature in staging.